subsys/bluetooth/host/conn.c

```diff
@@ -2655,6 +2677,14 @@ bt_security_t bt_conn_get_security(const struct bt_conn *conn)
 
 int bt_conn_cb_register(struct bt_conn_cb *cb)
```

Comment:
For the `bt_conn_cb_register` and `bt_conn_cb_unregister` I actually prefer the `#if defined` because then uses of that when `CONFIG_BT_CONN_DYNAMIC_CALLBACKS=n` will result in a compiletime error, rather than runtime, which I think is preferable to developers. Not sure if we have any specific requirements/guidelines on this though